openedx/core/djangoapps/programs/utils.py

```diff
@@ -461,57 +461,77 @@ def _attach_course_run_upgrade_url(self, run_mode):
     def _attach_course_run_may_certify(self, run_mode):
         run_mode['may_certify'] = self.course_overview.may_certify()
 
-    def _check_enrollment_for_user(self, course_run):
-        applicable_seat_types = self.data['applicable_seat_types']
 
-        (enrollment_mode, active) = CourseEnrollment.enrollment_mode_for_user(
-            self.user,
-            CourseKey.from_string(course_run['key'])
+    def _filter_out_courses_with_entitlements(self, courses):
+        course_uuids = set(course['uuid'] for course in courses)
+        entitlements = self.user.courseentitlement_set.filter(
+            course_uuid__in=set(course['uuid'] for course in courses),
+            mode__in=self.data['applicable_seat_types']
```

Comment:
I'm not sure if the mode lookup will work or not, as the `entitlements` from catalog have capitalized strings for mode, whereas the seat types in `applicable_seat_types` are not capitalized.